### DISCUSSION TITLE
Coverage improvements

### DISCUSSION THREAD

**Discussion by renefritze** [pymor/pymor#1510] [Ideas]
- looking at the [thermalblock_gui demo](https://app.codecov.io/gh/pymor/pymor/blob/main/src/pymordemos/thermalblock_gui.py) we should investigate why `ReducedSim._first` seems to be called not even once. Also see how difficult clicking some buttons with pytest-qt would be.
 - Add case that also hits the bottom `detailed_data` [branch](https://app.codecov.io/gh/pymor/pymor/blob/main/src/pymordemos/analyze_pickle.py)
 - `src/pymor/scripts/*` basically have no relevant coverage
 - [jupyter stuff ](https://codecov.io/gh/pymor/pymor/tree/main/src/pymor/discretizers/builtin/gui/jupyter) show very little coverage. They're used in tutorial tests though that do upload coverage data. So there's something amiss there.

**Comment by sdrave** [comment]
Actually, I'd say we can also remove `thermalblock_gui`. I do not think there is a point in maintaining the code. Rather we should invest some energy in having a `m.visualize` which build a similar gui for arbitrary `Models`.

**Reply by renefritze** [reply]
Right. Ofc that doesn't do away with the pytest-qt bit. Automagic GUIs need even more testing IMO.
